### PR TITLE
Use network helpers in the Hardhat Network docs

### PR DESCRIPTION
**Pull request by fvictorio**
Closes HH-915.

### CODE REVIEW

File: docs/src/content/hardhat-network-helpers/docs/reference.md [hunk 1]
```diff
@@ -268,3 +268,25 @@ it("test", async function () {
   // use token and exchanges contracts
 })
 ```
+
+## Other helpers
+
+### `dropTransaction(txHash)`
+
+Drops the given transaction hash from the mempool, if it exists.
```
Comment (alcuadrado) [inline docs/src/content/hardhat-network-helpers/docs/reference.md:276]:
```suggestion
Removes the given transaction from the mempool, if it exists.
```

The tx is dropped, not the hash

File: docs/src/content/hardhat-network-helpers/docs/reference.md [hunk 2]
```diff
@@ -268,3 +268,25 @@ it("test", async function () {
...
+
+Parameters:
+
+- `txHash`: Transaction hash to be dropped from mempool.
```
Comment (alcuadrado) [inline docs/src/content/hardhat-network-helpers/docs/reference.md:280]:
```suggestion
- `txHash`: Transaction hash to be removed from mempool.
```

File: docs/src/content/hardhat-network-helpers/docs/reference.md (hunk 1 above)
Comment (alcuadrado) [reply docs/src/content/hardhat-network-helpers/docs/reference.md:276]: also, `removed` is easier to understand

File: docs/src/content/hardhat-network/docs/explanation/mining-modes.md [hunk 3]
```diff
@@ -47,11 +47,11 @@ networks: {
 
 ## Removing and replacing transactions
 
-Transactions in the mempool can be removed using the `hardhat_dropTransaction` method:
+Transactions in the mempool can be removed using the [`dropTransaction`](</hardhat-network-helpers/docs/reference#droptransaction(txhash)>) network helper:
 
 ```js
 const txHash = "0xabc...";
-await network.provider.send("hardhat_dropTransaction", [txHash]);
+await helpers.dropTransaction(txHash);
```
Comment (alcuadrado) [inline docs/src/content/hardhat-network/docs/explanation/mining-modes.md:54]: `### Using RPC methods` still uses the provider.

**Comment by alcuadrado** [comment]
I think `hardhat_impersonateAccount` should be explained using both the helpers and ethers, in the forking guide.

File: docs/src/content/hardhat-network-helpers/docs/reference.md (hunk 1 above)
Comment (fvictorio) [reply docs/src/content/hardhat-network-helpers/docs/reference.md]: Done in d783f719d113df83fd5b3081d7ac0535ba5e43bf

File: docs/src/content/hardhat-network/docs/explanation/mining-modes.md (hunk 3 above)
Comment (fvictorio) [reply docs/src/content/hardhat-network/docs/explanation/mining-modes.md:54]: Yes, we don't have helpers to change the mining modes yet. [HH-921](https://linear.app/nomic-foundation/issue/HH-921/add-network-helper-to-configure-mining-behavior) tracks that.

**Comment by fvictorio** [comment]
> I think `hardhat_impersonateAccount` should be explained using both the helpers and ethers, in the forking guide.

What do you mean? For the record, this is the only mention to impersonating accounts we have right now in that guide:

image

I guess we could add a section about impersonating accounts before that section, wdyt?

**Comment by fvictorio** [comment]
Oh, you mean using `ethers.getImpersonatedSigner`? That makes sense, I will add that section.

File: docs/src/content/hardhat-network/docs/guides/forking-other-networks.md [hunk 4]
```diff
@@ -75,16 +75,30 @@ networks: {
...
+
+## Customizing Hardhat Network's behavior
 
-- [`hardhat_impersonateAccount`](../reference/#hardhat-impersonateaccount)
-- [`hardhat_stopImpersonatingAccount`](../reference/#hardhat-stopimpersonatingaccount)
-- [`hardhat_setNonce`](../reference/#hardhat-setnonce)
-- [`hardhat_setBalance`](../reference/#hardhat-setbalance)
-- [`hardhat_setCode`](../reference/#hardhat-setcode)
-- [`hardhat_setStorageAt`](../reference/#hardhat-setstorageat)
+Once you've got local instances of mainnet protocols, setting them in the specific state your tests need is likely the next step. For this, you can use our [Hardhat Network Helpers](/hardhat-network-helpers) library, which allows you to do things like manipulating the time of the network or modify the balance of an account.
```
Comment (feuGeneA) [inline docs/src/content/hardhat-network/docs/guides/forking-other-networks.md:101]:
```suggestion
Once you've got a local instance of the mainnet chain state, setting that state to the specific needs of your tests is likely the next step. For this, you can use our [Hardhat Network Helpers](/hardhat-network-helpers) library, which allows you to do things like manipulating the time of the network or modify the balance of an account.
```
The reason for this change was that I didn't like the use of the word `protocols`, but my attempts to change that one word had a sort of ripple effect.

**Review by feuGeneA — APPROVED** [review]

**Pull request merged**